src/main/webapp/sharing/sharing.js

```diff
@@ -0,0 +1,175 @@
+            permission,
+          }
+        })
+      }
+    })
+    .flat()
+    .filter(item => item !== undefined)
+}
+
+const convertState = (userState, groupState, metacardType) => {
+  const attributes = {
+    security_access_individuals_read: userState
+      .filter(user => user.permission === 'security_access_individuals_read')
+      .map(user => user.label),
+    security_access_individuals: userState
+      .filter(user => user.permission === 'security_access_individuals')
+      .map(user => user.label),
+    security_access_administrators: userState
+      .filter(user => user.permission === 'security_access_administrators')
+      .map(user => user.label),
+    security_access_groups_read: groupState
+      .filter(group => group.permission === 'security_access_groups_read')
+      .map(group => group.label),
+    security_access_groups: groupState
+      .filter(group => group.permission === 'security_access_groups')
+      .map(group => group.label),
+    metacard_type: metacardType,
+  }
+  return attributes
+}
+
+export default props => {
+  const { individuals, groups, userRoles, save, id, metacardType } = props
+  const [tabValue, setTabValue] = React.useState(0)
+  const [userState, setUserState] = React.useState(setupState(individuals))
+  const [groupState, setGroupState] = React.useState(setupState(groups))
+
+  const handleChange = (event, index, type, userOrGroup) => {
+    if (userOrGroup === 'user') {
+      const updatedUsers = [...userState]
+      updatedUsers[index][type] = event.target.value
+      setUserState(updatedUsers)
+    } else {
+      const updatedGroups = [...groupState]
+      updatedGroups[index][type] = event.target.value
+      setGroupState(updatedGroups)
+    }
+  }
+
+  const removeItem = (index, userOrGroup) => {
+    if (userOrGroup === 'user') {
+      const updatedUsers = [...userState]
+      updatedUsers.splice(index, 1)
+      setUserState(updatedUsers)
+    } else {
+      const updatedGroups = [...groupState]
+      updatedGroups.splice(index, 1)
+      setGroupState(updatedGroups)
+    }
+  }
+
+  const addItem = userOrGroup => {
+    userOrGroup === 'user'
+      ? setUserState([
+          ...userState,
+          { label: '', permission: 'security_access_individuals_read' },
+        ])
+      : setGroupState([
+          ...groupState,
+          { label: '', permission: 'security_access_groups_read' },
+        ])
+  }
+
+  return (
+    <React.Fragment>
+      <Tabs
+        value={tabValue}
+        onChange={(e, v) => {
+          setTabValue(v)
+        }}
+        variant="fullWidth"
+        scrollButtons="on"
+        indicatorColor="primary"
+        textColor="primary"
+      >
+        <Tab label="Users" icon={<Person />} />
+        <Tab label="Groups" icon={<Group />} />
+      </Tabs>
+      <Divider />
+      <Typography
+        component="div"
+        role="tabpanel"
+        hidden={0 !== tabValue}
+        id={`scrollable-force-tabpanel-${tabValue}`}
```

Comment:
Can we leave a comment for what this id does?

Reply:
Removed